### PR TITLE
feature/86ep3ht47/Bowtie-Drag-n-Drop

### PR DESCRIPTION
**Pull request by acostales08**

### CODE REVIEW

**Review by JezChristian — APPROVED** [review]
Galing!!!!

File: apps/simulator/db.json [hunk 1]
```diff
@@ -668,6 +668,117 @@
...
+                        "answer": [
+                            {
+                                "answerId":9,
+                                "qId":9,
+                                "answerType":"DNDBowtie",
+                                "answerInstruction": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique quasi obcaecati laudantium quibusdam aspernatur esse ad totam dolorum?",
+                                "answerUI": "Table",
+                                "columns": ["Body System", "Potential Nursing Interventions"],
+                                "rows": [],                            
+                                "selectFieldKey": [
```
Comment (JMSevilla) [inline apps/simulator/db.json:696]: This should not be `selectFieldKey` change the property name according to the questionnaire.

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/DraggableCard.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { Paper, IconButton } from "@mui/material";
+import { useDrag } from "react-dnd";
+import HighlightOffOutlinedIcon from '@mui/icons-material/HighlightOffOutlined';
+import { dndObjectValueProps } from '@/core/types/ssrData';
+
+type Props = {
+    onClick?: (() => void) | undefined
+    answer: dndObjectValueProps, 
+    bg?: any,
+    icon?: boolean | undefined
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/DraggableCard.tsx:11]: Why we are passing icon as props? icon should not be an optional it is always included in your component.

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/Droppable.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+type Props = {
+    accept: string[],
+    text: string,
+    bg?: any,
+    onDrop: (item: dndObjectValueProps) => void,
+    droppedValue: dndObjectValueProps[],
+    handleRemove: () => void
+}
+
+export const DroppableContainer: React.FC<Props> = ({accept, text, bg, onDrop, droppedValue, handleRemove }) => {
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/Droppable.tsx:16]: avoid prop drilling, you can use ...others or ...rest

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/DraggableCard.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    type: string
+}
+
+export const DraggableCard: React.FC<Props> = ({ onClick, answer, bg, icon, type}) => {
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/DraggableCard.tsx:15]: avoid prop drilling, you can use ...others or ...rest

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/Droppable.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    drop(ref);
+
+    return (
+        <div ref={ref} className={`flex justify-center items-center text-center text-sm w-48 h-16 p-2 rounded-md shadow-inner ${bg} ${isOver ? 'bg-gray-200' : ''}`}>
+            {droppedValue.length > 0 ? (
+                <>
+                    <DraggableCard answer={droppedValue[0]} bg={bg} icon={true} onClick={handleRemove} type="sample"/>
+                </>
+            ) : (
+                <p className='opacity-35'>{text}</p>
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/Droppable.tsx:35]: Is this kind of catch ? is your `droppedValue` can possibly be empty? if not you can use `droppedValue.length > 0 &&`

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/ItemTypes.tsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+const itemTypes = {
+    FIRST: "Action To Take",
+    SECOND: "Potential Condition",
+    Third: "Parameters To Monitor"
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/ItemTypes.tsx:4]: Inconsistent casing.

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/ItemTypes.tsx [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+  };
+  
+  export default itemTypes;
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieComponent/ItemTypes.tsx:7]: using export default is for classes only. you can use `export const itemTypes` 
[suggestion] create a folder in core folder named `constants`

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieQuestionaire.tsx [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+import { Grid, Paper, Typography } from '@mui/material';
+import NearMeIcon from '@mui/icons-material/NearMe';
+import { DroppableContainer } from './DNDBowtieComponent/Droppable';
+import { DraggableCard } from './DNDBowtieComponent/DraggableCard';
+import itemTypes from './DNDBowtieComponent/ItemTypes';
+import { dndObjectValueProps, SsrData } from '@/core/types/ssrData';
+
+
+const initialContainerState= {
+  "first": [],
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieQuestionaire.tsx:11]: your containerState should have a proper naming convension. (e.g., ActionToTake1, ActionToTake2) with this naming convention we can easily identify your containerState

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieQuestionaire.tsx [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+};
+
+export const DNDBowtieQuestion: React.FC<SsrData> = ({ questionaire, answer }) => {
+
+    const foundAnswer = answer ? answer && answer.find(answers => answers.answerId) : null;
+    const choicesList = foundAnswer ? foundAnswer.ChoicesList : [];
+
+const [answerLists, setAnswerList] = useState<Record<string, dndObjectValueProps[]>>(choicesList);
+  const [droppedValue, setDroppedValue] = useState<Record<string, dndObjectValueProps[]>>(initialContainerState);
+  const [dropContainer] = useState([
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieQuestionaire.tsx:25]: Data is too messy you can transfer those in constants.

File: apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieQuestionaire.tsx [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+                            ))}
+                        </div>
+                    </div>
+                    </Paper>
+                    <Paper elevation={3} className="p-1 overflow-auto flex flex-col">
+                      <div className="flex justify-evenly items-start">
+                        <div className="flex gap-1">
+                          {answerItem.selectFieldKey.map((key: string, index: number) => (
+                            <div key={index} className="min-w-[180px] bg-[#E6F2FF]">
+                              {/* <h3 className="text-center p-2">{key}</h3> */}
```
Comment (JMSevilla) [inline apps/simulator/src/components/blocks/CaseStudy/CaseStudyQuestions/DNDBowtieQuestionaire.tsx:167]: remove

File: apps/simulator/src/core/types/ssrData.ts [hunk 11]
```diff
@@ -41,11 +41,12 @@ export type AnswerProps = {
   DDCAnswer: string;
   selectFieldKey: string[];
   selectField: any;
+  ChoicesList: any
```
Comment (JMSevilla) [inline apps/simulator/src/core/types/ssrData.ts:44]: Can we identify the types for this instead of any ?

**Review by JMSevilla — CHANGES_REQUESTED** [review]

**Review by JMSevilla — CHANGES_REQUESTED** [review]
Please address all the comments

**Review by DropitupQwerty — APPROVED** [review]

**Review by JMSevilla — APPROVED** [review]

**Pull request merged**